### ISSUE TITLE
MiniStarter.open() hangs indefinitely

### ISSUE THREAD

**Issue by dtomvan** [echasnovski/mini.nvim#186]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.starter

### Description

When opening the `mini.starter`, either on `VimEnter` or manually, on the default settings, Neovim hangs irrecoverably.

Probably caused by 32cb2b23805a?

### Neovim version

Neovim nightly v0.9.0-dev-594+gc590641fe commit neovim/neovim@c590641febf4d03e89c46f8e7ef4c3fb2a455520

### Steps to reproduce

Minimal `init.lua`:
```lua
local lazypath = vim.fn.stdpath 'data' .. '/lazy/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system {
        'git',
        'clone',
        '--filter=blob:none',
        '--single-branch',
        'https://github.com/folke/lazy.nvim.git',
        lazypath,
    }
end
vim.opt.runtimepath = { lazypath }

require('lazy').setup { {
    'echasnovski/mini.nvim',
    config = function()
        require 'mini.starter'.setup {}
    end,
} }
```
1. `nvim -nu minimal.lua`

-- OR --

```lua
require('lazy').setup {
    'echasnovski/mini.nvim',
}
require'mini.starter'.setup {
    autoopen = false,
}
```

1. `nvim -nu minimal.lua`
2. `:lua MiniStarter.open()`

### Expected behavior

MiniStarter opens normally with the default settings

### Actual behavior

Neovim hangs completely, even doing a Ctrl-C doesn't help.

**Comment by dtomvan** [comment]
Hmm, I cannot reproduce on a different machine. Could it have something to do with v:oldfiles, since that is the only "dynamic" content in the default settings?

**Comment by echasnovski** [comment]
I can not reproduce this also with your Neovim version and built-in package manager.

I see you use both the latest Neovim version and 'folke/noice.nvim'. Can it be [this issue](https://github.com/folke/noice.nvim/issues/298)?

**Comment by echasnovski** [comment]
> I can not reproduce this also with your Neovim version and built-in package manager.

I also tried it with your Neovim version and 'lazy.nvim'. Could not reproduce. Maybe still you have 'noice.nvim' loaded somewhere?

**Comment by folke** [comment]
I don't think this is the noice issue, since that would simply crash Neovim and not hang.

**Comment by echasnovski** [comment]
@folke, yeah this might be the case. I just saw newest Neovim version and remembered this 'noice.nvim' issue. Probably, they both have similar upstream root cause.

@dtomvan, since I can not reproduce this, when you are back at the original machine could you try the following:
- Latest 'mini.starter' but earlier Neovim version (0.8.2 or at most https://github.com/neovim/neovim/commit/99cf111289bfcd14981255e805da43bac5139141). If issue goes away, then it is a Neovim nightly release. Will wait for some time and see progress on Neovim. Ideally, finding minimal reproducible example and filing an upstream issue is better. 
- Latest Neovim but earlier 'mini.starter' version (before 32cb2b23805a52c75795792a2076a46a8d235486). If issue goes away, could you try replacing [this line](https://github.com/echasnovski/mini.nvim/blob/06eddfd8d6341e6c4f53ec9ae109a376a566329d/lua/mini/starter.lua#L1284) with `vim.schedule(function() vim.api.nvim_input([[<C-\><C-n>]]) end)` and try again?

**Comment by dtomvan** [comment]
> I see you use both the latest Neovim version and 'folke/noice.nvim'.

@echasnovski When I'm using the minimal init.lua (I also tried it with packer.nvim, just to be sure) I'm only installing mini.starter, right?

**Comment by dtomvan** [comment]
Huh. After rebooting (next use after yesterday) of the machine I reported this issue on, it just _went away_. I am sorry for bothering you with this. I have no idea how the issue might've came up, since I just can't reproduce it at all anymore.

**Comment by dtomvan** [comment]
All I can think of is maybe some spare files left off (in `/tmp`) by Neovim's new TUI-rework. Those might've made some Neovim API function hang, which in turn was called by mini.starter.

Also, I forgot to mention, I've even went back to the first occurence in git history of `lua/mini/starter.lua` and even that revision had the problem, so it's definitely not an issue with your code.

**Comment by echasnovski** [comment]
> @echasnovski When I'm using the minimal init.lua (I also tried it with packer.nvim, just to be sure) I'm only installing mini.starter, right?

Should be true, yes. I am yet to study the difference between how 'lazy.nvim', 'packer.nvim', and built-in packages find what to load. There might be cases that plugin was installed in some place from where user forgot to remove it, thus it was loaded even when not asking explicitly. So it was just to be sure.

> Huh. After rebooting (next use after yesterday) of the machine I reported this issue on, it just went away. I am sorry for bothering you with this. I have no idea how the issue might've came up, since I just can't reproduce it at all anymore.

No problem. There is some increased instability in Neovim nightly, though. I just tried to use it to run some tests and it indeed shows some "hanging" behavior. Interactively it works almost without changes. So yeah, I would restrain from using Neovim nightly for some time.

Closing this, then. If you ever encounter it again, please, try to find reproducible steps and feel free to post/reopen.
